File: 02-download-a-file/java/src/test/java/Download.java
package src.test.java;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.By;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.firefox.FirefoxProfile;
import org.openqa.selenium.firefox.FirefoxOptions;
import static org.hamcrest.CoreMatchers.*;
import static org.junit.Assert.assertThat;
import java.io.File;
import java.util.UUID;


public class Download {
    WebDriver driver;
    File folder;

    @Before
    public void setUp() throws Exception {
        folder = new File(UUID.randomUUID().toString());
        folder.mkdir();

        FirefoxOptions options = new FirefoxOptions();
        FirefoxProfile profile = new FirefoxProfile();
        profile.setPreference("browser.download.dir", folder.getAbsolutePath());
        profile.setPreference("browser.download.folderList", 2);
        profile.setPreference("browser.helperApps.neverAsk.saveToDisk",
                "image/jpeg, application/pdf, application/octet-stream");
        profile.setPreference("pdfjs.disabled", true);
        options.setProfile(profile);
        driver = new FirefoxDriver(options);

    }

    @After
    public void tearDown() throws Exception {
        driver.quit();
        for (File file: folder.listFiles()) {
            file.delete();
        }
        folder.delete();
    }

    @Test
    public void download() throws Exception {
        driver.get("http://the-internet.herokuapp.com/download");
        driver.findElement(By.cssSelector(".example a")).click();
        // Wait 2 seconds to download file
        Thread.sleep(2000);
        File[] listOfFiles = folder.listFiles();
        // Make sure the directory is not empty
        assertThat(listOfFiles.length, is(not(0)));
        for (File file : listOfFiles) {
            // Make sure the downloaded file(s) is(are) not empty
            assertThat(file.length(), is(not((long) 0)));
        }
    }

}

File: 02-download-a-file/kotlin/src/test/kotlin/Download.kt
import org.openqa.selenium.By
import org.openqa.selenium.firefox.FirefoxDriver
import org.openqa.selenium.firefox.FirefoxProfile
import java.io.File
import java.util.*
import java.util.concurrent.TimeUnit
import kotlin.test.Test
import kotlin.test.AfterTest
import kotlin.test.assertNotEquals
import org.openqa.selenium.firefox.FirefoxOptions

// 02-download-a-file
class Download {
    private val testUrl = "http://the-internet.herokuapp.com/download"
    private val folder = File(UUID.randomUUID().toString())

    private val driver by lazy {
        folder.mkdir()
        val profile = FirefoxProfile()
        profile.setPreference("browser.download.dir", folder.absolutePath)
        profile.setPreference("browser.download.folderList", 2)
        profile.setPreference(
            "browser.helperApps.neverAsk.saveToDisk",
            "image/jpeg, application/pdf, application/octet-stream"
        )
        profile.setPreference("pdfjs.disabled", true)
        val options = FirefoxOptions().setProfile(profile)
        FirefoxDriver(options)
    }

    @AfterTest
    fun tearDown() {
        // Quit session (closes browser)
        driver.quit()
        for (file in folder.listFiles()!!) {
            file.delete()
        }
        folder.delete()
    }

    @Test
    @Throws(Exception::class)
    fun download() {
        driver.get(testUrl)
        driver.findElement(By.cssSelector(".example a")).click()

        // Implicit wait
        driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS)

        val listOfFiles = folder.listFiles()

        // Make sure the directory is not empty
        assertNotEquals(
            listOfFiles!!.size,
            0
        )
        for (file in listOfFiles) {
            // Make sure the downloaded file(s) is(are) not empty
            assertNotEquals(
                file.length(),
                0
            )
        }
    }
}